test/irqbalance.bats

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env bats
+# vim: set syntax=sh:
+
+load helpers
+
+function detect_irqbalance_config() {
+	# debian/ubuntu
+	[ -f /etc/default/irqbalance ] && echo "/etc/default/irqbalance"
+	# fedora/centos/RHEL
+	[ -f /etc/sysconfig/irqbalance ] && echo "/etc/sysconfig/irqbalance"
+	# default
+	echo ""
+}
+
+function setup_file() {
+	if ! command -v irqbalance > /dev/null; then
+		skip "irqbalance not found."
+	fi
+	IRQBALANCE_CONF=$(detect_irqbalance_config)
+	if [ -z "$IRQBALANCE_CONF" ]; then
+		echo "error: unable to find irqbalance config file"
+		return 1
+	fi
+	CONFIGLET="$CRIO_CONFIG_DIR/99-irqbalance.conf"
+
+	mkdir -p "/etc/sysconfig"
+	BANNEDCPUS_CONF="/etc/sysconfig/orig_irq_banned_cpus"
+
+	export IRQBALANCE_CONF
+	export CONFIGLET
+	export BANNEDCPUS_CONF
+}
+
+function setup() {
+	setup_test
+	stop_crio
+	# we don't uncondintionally restore because the irqbalance package may be missing
+	# from the test system. if this is the case, one less thing to worry about, we can't
+	# pollute the system state anyway :)
+	if [ -f "$IRQBALANCE_CONF" ]; then
+		cp -v "$IRQBALANCE_CONF" "$IRQBALANCE_CONF".bkp
+		expected_banned_cpus=$(sed -n 's/^IRQBALANCE_BANNED_CPUS=\"\?\([^\"]*\)\"\?/\1/p' "$IRQBALANCE_CONF")
+		if [ -z "$expected_banned_cpus" ]; then
+			echo "warning: unable to read 'IRQBALANCE_BANNED_CPUS' from ${IRQBALANCE_CONF}. Creating default value:0"
```

Comment:
do we need a default value here? can't it be empty string? (aka no banned CPU)

Reply:
It is not extrictly needed, but some tests check for the value before and after execution and having a default value could help. 
I do not have a strong opinion about it so, I could get rid of it if you think it is not needed.

Reply:
the default content of the file in RHEL/Fedora is empty banned cpus, and we should make sure this flow is covered. The default originally added were ment to cover this flow